src/ast/code_generator.ml

```diff
@@ -199,12 +176,12 @@ let rec term_generate tenv term : (binary expr * binary val_type) Result.t =
     let* expr2, ty2 = term_generate tenv tm2 in
     binop_generate b expr1 ty1 expr2 ty2
   | Result (Some i) -> (
-    match tenv#get_result_type i with
-    | Some t -> Ok ([ Local_get (Raw (tenv#get_param_number + i)) ], t)
+    match List.nth_opt tenv.result_types i with
```

Comment:
If you're using `nth{_opt}` then you should probably replace this type by an array.

Reply:
Here (along with 3 other `List.nth_opt`s), the index may be out of bound. Is it okay to explicitly catch the `Invalid_argument` exception when using array?

Reply:
Oh maybe check the bound first, I'll change it for an array